sheepdog/transactions/upload/sub_entities.py

```diff
@@ -531,7 +532,10 @@ def get_file_from_index_by_hash(self):
             # if `document` exists, `document.did` is the UUID that is already
             # registered in indexd for this entity.
             if params:
-                document = self.transaction.signpost.get_with_params(params)
+                try:
+                    document = self.transaction.signpost.get_with_params(params)
+                except requests.HTTPError as e:
+                    raise UserError(code=e.response.status_code, message=e.message)
```

Comment:
we should add some context to this error message:
`Fail to register this data node in indexd: {message}` ?